CHANGELOG.md

```diff
@@ -27,7 +27,7 @@
 * Support sort/distinct based on values from a dictionary e.g. `TRUEPREDICATE SORT(meta['age'])` (PR [#5311](https://github.com/realm/realm-core/pull/5311))
 
 ### Fixed
-* Fix the query parser needs to copy list of arguments and own the memory. ([#6674](https://github.com/realm/realm-core/pull/6674), since v12.5.0)
+* Fix the query parser, it needs to copy a list of arguments and own the memory. This will prevent errors like getting a different result from a query, if the list is modified after its creation and before the execution of the query itself. In the worst case scenario, if the memory is freed before the query is executed, this could lead to crashes, especially for string and binary data types. ([#6674](https://github.com/realm/realm-core/pull/6674), since v12.5.0)
```

Comment:
Have we seen reports of this?
I would delete the first sentence which is about how it was fixed and focus on describing the problem in the form that's in the template:
> * <How do the end-user experience this issue? what was the impact?>

i.e. When <If a user did this> <then this could/would happen> <refer to issues where that has been observed or say that it hasn't been reported by users>

Reply:
I don't think it is easy to explain this in terms of how the final user can experience the issue. Since this API is consumed by the SDKs, also consumers of the C++ API are less likely to notice any issues since they do not need to keep track of the memory allocated as much as the C-API consumers do. 
In fact, we haven't seen reports of this, but as soon as Kotlin started to support lists in the queries, this issue was discovered by @rorbech